pkg/apis/networking/v1alpha1/certificate_validation_test.go

```diff
@@ -36,27 +36,46 @@ func TestCertificateSpecValidation(t *testing.T) {
 			SecretName: "secret",
 		},
 		want: nil,
+	}, {
+		name: "valid-with-domain",
+		cs: &CertificateSpec{
+			DNSNames:   []string{"host.example"},
```

Comment:
Not strictly necessary; you could add `"host.foo"` to DNSNames to test both paths through the loop, but I'm pretty confident with the coverage overall.